src/components/gift_order_page/ProductInfo.tsx

```diff
@@ -79,39 +81,62 @@ const PriceValue = styled.div`
 `;
 
 export const ProductInfo = () => {
+  const navigate = useNavigate();
   const { id } = useParams();
   if (!id) throw new Error('id가 없습니다');
   const parsedId = parseInt(id!);
-  const currentGift = giftItemData[parsedId];
+  const [currentGift, setCurrentGift] = useState<GiftItemDataType>();
   const { setId, setName, setPrice, setBrand } = useProductInfo();
 
   useEffect(() => {
+    const getData = async () => {
+      try {
+        const response = await apiClient.get(`/api/products/${parsedId}`);
+        setCurrentGift(response.data.data);
+      } catch (error) {
+        console.log('⚠️ 요청 처리 중 오류가 발생했습니다.', error);
+        toast.warn('⚠️ 로그인 요청 처리 중 오류가 발생했습니다.', {
+          position: 'bottom-center',
+          autoClose: 5000,
+          hideProgressBar: false,
+          closeOnClick: false,
+          pauseOnHover: true,
+          draggable: true,
+          progress: undefined,
+          theme: 'dark',
+          transition: Bounce,
+          style: {
+            width: '25rem',
+            color: 'black',
+            backgroundColor: 'white',
+          },
+        });
+        navigate('/');
+      }
+    };
+    getData();
+  }, [navigate, parsedId]);
+
+  useEffect(() => {
+    if (!currentGift) return;
+
     setId(currentGift.id);
     setName(currentGift.name);
     setPrice(currentGift.price.basicPrice);
     setBrand(currentGift.brandInfo.name);
-  }, [
-    currentGift.brandInfo.name,
-    currentGift.id,
-    currentGift.name,
-    currentGift.price.basicPrice,
-    setBrand,
-    setId,
-    setName,
-    setPrice,
-  ]);
+  }, [currentGift, setBrand, setId, setName, setPrice]);
 
   return (
     <Container>
       <Label>상품 정보</Label>
       <Body>
-        <ProductImg src={currentGift.imageURL} />
+        <ProductImg src={currentGift?.imageURL} />
```

Comment:
imageUrl이 없는 경우에는 조건부로 이미지를 렌더링해주거나, 기본 이미지를 넣어주시는건 어떨까요?

```suggestion
       {currentGift?.imageURL && <ProductImg src={currentGift.imageURL} /> }
```

```suggestion
       <ProductImg src={currentGift?.imageURL || '기본이미지URL'} />
```